CA_DataUploaderLib/IOconf/IOconfFilter.cs

```diff
@@ -26,20 +33,96 @@ public IOconfFilter(string row, int lineNum) : base(row, lineNum, "Filter")
             if (!list[3].TryToDouble(out filterLength))
                 throw new Exception($"Wrong filter length: {row} {Environment.NewLine}{Format}");
 
+            if (!IsDecisionFilter)
+                (SourceNames, HideSource) = InitLegacyFilter(list);
+            else if (list.Count != 5)
+                throw new FormatException($"Unexpected format for {filterType} filter: {Row}{Environment.NewLine}{Format}");
+            else
+                (_sustainedExpression, SourceNames) = InitSustainedFilter(list, Row, Format);
+        }
+
+        private (List<string> sources, bool hideSource) InitLegacyFilter(List<string> list)
+        {
+            var hideSource = false;
             var sources = list.Skip(4).ToList();
-            if (sources.Last() == "hidesource")
+            if (!IsDecisionFilter && sources.Last() == "hidesource")
             {
-                HideSource = true;
+                hideSource = true;
                 sources.Remove("hidesource");
             }
 
             // source validation happens later, as there is not a 1-1 relation of IOConfFile entries and values getting into the Vector i.e. some oxygen sensors have 3 values. 
-            SourceNames = sources;
+            return (sources, hideSource);
+        }
+
+        private static (LogicalExpression sustainedExpression, List<string> sourceNames) InitSustainedFilter(List<string> list, string row, string format)
+        {
+            try
+            {
+                var (sustainedExpression, sourceNames) = IOconfMath.CompileExpression(list[4]);
+                // Perform test calculation using default input values
+                if (IOconfMath.Calculate(sourceNames.ToDictionary(s => s, s => (object)0), sustainedExpression) is not bool)
+                    throw new FormatException($"Only boolean filter expressions are supported: {row}{Environment.NewLine}{format}");
+                return (sustainedExpression, sourceNames);
+            }
+            catch (OverflowException ex)
+            {
+                throw new OverflowException($"Filter expression causes integer overflow: {row}{Environment.NewLine}{format}", ex);
+            }
+            catch (FormatException) { throw; }
+            catch (Exception ex)
+            {
+                throw new Exception($"Wrong format for filter expression: {row}{Environment.NewLine}{format}", ex);
+            }
         }
 
         public override IEnumerable<string> GetExpandedSensorNames(IIOconf ioconf)
         {
             yield return NameInVector;
         }
+
+        public IEnumerable<string> GetDecisionFields() => 
+            filterType == FilterType.Sustained 
+                ? [NameInVector, NameInVector + "_targettime"]
+                : throw new InvalidOperationException($"Unexpected call to GetDecisionFields for filter type {Row}");
```

Comment:
"... for filter type {_filterType_}"?

Reply:
It would be a bug, so I would rather have the whole row.  Changing it to this, but we can talk if that's still not ok: Unexpected call to GetDecisionFields in {Row}